src/enzo/ActiveParticle_Skeleton.C

```diff
@@ -0,0 +1,309 @@
+
+  // This is here for compatibility with old-style ascii parameter files, where
+  // these parameters exist as global variables that are assigned in
+  // ReadParameterFile using sscanf.
+  OverdensityThreshold = StarMakerOverDensityThreshold;
+  MassEfficiency = StarMakerMassEfficiency;
+
+
+  AttributeVector &ah = ActiveParticleType_Skeleton::AttributeHandlers;
+
+  // This sets up the particle attributes that are defined on the base
+  // ActiveParticleType class.  This includes thing like mass, position,
+  // velocity, unique ID, etc.
+  ActiveParticleType::SetupBaseParticleAttributes(
+      ActiveParticleType_Skeleton::AttributeHandlers);
+
+  // Register instance member variables here.  Warning: if you do not do this,
+  // this data will not be saved to disk or communicated over MPI.
+  ah.push_back(new Handler<ActiveParticleType_Skeleton, float,
+      &ActiveParticleType_Skeleton::AccretionRate>("AccretionRate"));
+
+  return SUCCESS;
+}
+
+int ActiveParticleType_Skeleton::EvaluateFormation(grid *thisgrid_orig, ActiveParticleFormationData &data)
+{
+  // Create a 'friend' grid alias that we can use to access private grid data.
+  SkeletonGrid *thisGrid =
+    static_cast<SkeletonGrid *>(thisgrid_orig);
+
+  int i, j, k, dim, index, offset_y, offset_z;
+  int NumberOfNewParticles = 0;
+
+  float *density = thisGrid->BaryonField[data.DensNum];
+  float *velx = thisGrid->BaryonField[data.Vel1Num];
+  float *vely = thisGrid->BaryonField[data.Vel2Num];
+  float *velz = thisGrid->BaryonField[data.Vel3Num];
+  float *totalenergy = thisGrid->BaryonField[data.TENum];
+  float *gasenergy = thisGrid->BaryonField[data.GENum];
+  float *metals = thisGrid->BaryonField[data.MetalNum];
+  float dt = thisGrid->dtFixed;
+  FLOAT dx = data.LengthUnits * thisGrid->CellWidth[0][0];
+
+  FLOAT CurrentTime = thisGrid->Time;
+  FLOAT xstart = thisGrid->CellLeftEdge[0][0];
+  FLOAT ystart = thisGrid->CellLeftEdge[1][0];
+  FLOAT zstart = thisGrid->CellLeftEdge[2][0];
+
+  bool HasMetalField = (data.MetalNum != -1 || data.ColourNum != -1);
+
+  int NumberOfGhostZones = thisGrid->GridStartIndex[0];
+  int GridDimension[3] = {thisGrid->GridDimension[0],
+                          thisGrid->GridDimension[1],
+                          thisGrid->GridDimension[2]};
+
+  // Pre-calculate serialized offsets for the 3D data field.  Used for
+  // the divergence.
+  offset_y = thisGrid->GridDimension[0];
+  offset_z = thisGrid->GridDimension[0] * thisGrid->GridDimension[1];
+
+  float StarFraction = 0.1;
+  float DynamicalTime = -1;
+
+  for (k = thisGrid->GridStartIndex[2]; k <= thisGrid->GridEndIndex[2]; k++)
+  {
+    for (j = thisGrid->GridStartIndex[1]; j <= thisGrid->GridEndIndex[1]; j++)
+    {
+      index = GRIDINDEX_NOGHOST(thisGrid->GridStartIndex[0], j, k);
+      for (i = thisGrid->GridStartIndex[0]; i <= thisGrid->GridEndIndex[0]; i++, index++)
+      {
+
+        // Physics goes here.
+
+        if (false)
```

Comment:
Is the block of code after "if(false)" still necessary?

Reply:
The block of code is necessary as an example of how to write an ActiveParticle. I think the `if(false)` should be `if(true)` though so I've changed it. It doesn't matter of course since this is just template code - it never runs.